x/launchdarkly/flags/evaluationcontext/user.go

```diff
@@ -73,6 +74,30 @@ func NewAnonymousUser(key string) User {
 	return u
 }
 
+// NewAnonymousUserWithSubdomain returns a user object suitable for use in unauthenticated with known subdomain
+// requests or requests with no access to user identifiers.
+// Provide a unique session or request identifier as the key if possible. If the
+// key is empty, it will default to an uuid so percentage rollouts will still apply.
+// No userID will be given to an anonymous user.
+func NewAnonymousUserWithSubdomain(key string, subdomain string) User {
+	if key == "" {
+		key = uuid.NewString()
+	}
+
+	u := User{
+		key: key,
+	}
+
+	userBuilder := lduser.NewUserBuilder(u.key)
+	userBuilder.Anonymous(true)
+	userBuilder.Custom(
+		userAttributeSubdomain,
+		ldvalue.String(subdomain))
+	u.ldUser = userBuilder.Build()
+
+	return u
```

Comment:
This can be simplified.

```suggestion
	return User{
		key: key,
		ldUser: lduser.NewUserBuilder(key).
			Anonymous(true).
			Build(),
	}
```